containerd/main.go

```diff
@@ -94,17 +101,41 @@ func daemon(address, stateDir string, concurrency int, runtimeName string) error
 	if err := sv.Start(); err != nil {
 		return err
 	}
-	if err := os.RemoveAll(address); err != nil {
+	server, err := startServer(address, sv)
+	if err != nil {
 		return err
 	}
+	for ss := range s {
+		switch ss {
+		case syscall.SIGCHLD:
+			if _, err := osutils.Reap(); err != nil {
+				logrus.WithField("error", err).Warn("containerd: reap child processes")
+			}
+		default:
+			server.Stop()
```

Comment:
Can we have an info log saying that the server is stopping because it received signal X?